### PR TITLE
Fix #346: Add support for inline radios and checkboxes.

### LINKED ISSUE #346: Inline checkboxes and radios
### Prerequisites

- [x] I have read the [documentation](https://github.com/mozilla-services/react-jsonschema-form/blob/master/README.md#readme).

### Description

I would like to display inputs with `type="radio"` or `type="checkbox"` with [inline markup from Bootstrap](http://getbootstrap.com/css/#inline-checkboxes-and-radios). 

Is there a way to add option like `{ "ui:inline": true }` to enable this flag or I have to override the whole widget?    

### Version

0.40.0

### PR DESCRIPTION
**Pull request by n1k0**
This patch enables having inline checkboxes and radio buttons. 

demo

- [x] List of checkboxes
- [x] List of radio buttons
- [x] Boolean field radio buttons case
- [x] Docs
- [x] Tests

### CODE REVIEW

**Comment by n1k0** [comment]
r=? @leplatrem

File: src/components/fields/BooleanField.js [hunk 1]
```diff
@@ -4,13 +4,17 @@ import {
...
 } from "../../utils";
 import CheckboxWidget from "./../widgets/CheckboxWidget";
 
 
-function buildOptions(schema) {
+function buildOptions(schema, uiSchema={}) {
   return {
+    inline: isObject(uiSchema) &&
+            isObject(uiSchema.options) &&
+            uiSchema.options.inline,
```
Comment (leplatrem) [inline src/components/fields/BooleanField.js]: `const {uiSchema: {options: {inline}={}};` ?

File: src/components/widgets/CheckboxesWidget.js [hunk 2]
```diff
@@ -15,13 +15,13 @@ function deselectValue(value, selected) {
...
   const {id, disabled, options, value, autofocus, onChange} = props;
-  const {enumOptions} = options;
+  const {enumOptions, inline} = options;
   return (
     <div className="checkboxes" id={id}>{
       enumOptions.map((option, index) => {
         const checked = value.indexOf(option.value) !== -1;
         return (
-          <div key={index} className="checkbox">
+          <div key={index} className={inline ? "checkbox-inline" : "checkbox"}>
```
Comment (leplatrem) [inline src/components/widgets/CheckboxesWidget.js]: `checkbox${inline ? "-inline" : ""}` ?

File: src/components/fields/BooleanField.js (hunk 1 above)
Comment (n1k0) [inline src/components/fields/BooleanField.js]: I'm not sure this is valid syntax (unfortunately)

Comment (n1k0) [inline src/components/fields/BooleanField.js]: Yeah, dunno why my sublime eslint plugin shows me this in error though :(

Anyway, the thing is the argument can be differently typed, so asuming it's always a destructurable object is highly error prone methinks (see added comments in https://github.com/mozilla-services/react-jsonschema-form/pull/348/commits/6063243fc613684a38193280080899c300f88085).

**Comment by n1k0** [comment]
Released in [v0.41.0](https://github.com/mozilla-services/react-jsonschema-form/releases/tag/v0.41.0).

**Pull request merged**